pkg/controllermanager/controller/shoot/shoot_maintenance_control.go

```diff
@@ -142,11 +142,15 @@ func (r *shootMaintenanceReconciler) reconcile(ctx context.Context, shoot *garde
 		return err
 	}
 
-	updatedMachineImages, reasonForImageUpdatePerPool, err := MaintainMachineImages(shootLogger, shoot, cloudProfile)
+	reasonForImageUpdatePerPool, err := MaintainMachineImages(shootLogger, shoot, cloudProfile)
 	if err != nil {
 		// continue execution to allow the kubernetes version update
 		shootLogger.Error(fmt.Sprintf("Could not maintain machine image version: %s", err.Error()))
 	}
+	for _, reason := range reasonForImageUpdatePerPool {
```

Comment:
this link doesn't work for me, I assume you mean [this](https://github.com/gardener/gardener/blob/c71c50b53bb7de36a441d9d4df43cb1dd6ac89ae/pkg/controllermanager/controller/shoot/shoot_maintenance_control.go#L203) point, just like the k8s version upgrade reasons are recorded?

Reply:
Yep, that's what I meant.

Reply:
I've moved the entire block down, event  recording as well as logging. BTW: does it make sense to have the same string logged *and* event recorded?

Reply:
> BTW: does it make sense to have the same string logged and event recorded?

Hm, events are volatile while we save the Gardenlet logs for a longer period. So if we ran a Gardenlet with `debug` log level and want to look up this message hours/days later, then it would be beneficial. But in this particular case, I doubt that it makes too much sense, tbh.

Reply:
So should I be removing the duplicated log output or do we want to leave it like it is now?

Reply:
No strong opinion from my side.